shared/features/social-link/social-link-tag/social-link-tag.tsx

```diff
@@ -1,19 +1,41 @@
+import copy from "copy-to-clipboard";
+import { useMemo } from "react";
+
 import { Button } from "@/design-system/atoms/button/variants/button-default";
+import { Icon } from "@/design-system/atoms/icon";
+
+import { SOCIAL_PLATFORM } from "@/shared/constants/social-platform";
+import { Discord, Whatsapp } from "@/shared/icons";
 
 import { SocialIconLink } from "../social-icon-link/social-icon-link";
 import { SocialLinkTagProps } from "./social-link-tag.types";
 
-export function SocialLinkTag({ url, value }: SocialLinkTagProps) {
-  if (!value) return null;
+export function SocialLinkTag({ url, value, channel }: SocialLinkTagProps) {
+  const args = useMemo(() => {
+    if (channel === SOCIAL_PLATFORM.WHATSAPP) {
+      return {
+        onClick: () => copy(url),
```

Comment:
Add a toaster to say that it has been copied?

Reply:
i did it because i like the idea ^^